### PR TITLE
Supporting UUIDs for /subscriptions resource

### PR DESCRIPTION
**Pull request by malinthaprasan**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/SubscriptionsApiServiceImpl.java [hunk 1]
```diff
@@ -28,18 +25,27 @@
     @Override
     public Response subscriptionsGet(String apiId, String applicationId, String groupId, String accept,
             String ifNoneMatch) {
-
+        //todo: validation: only one of {application id,api id} should present
```
Comment (kasunbg) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/SubscriptionsApiServiceImpl.java]: Can you fix this TODO?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 2]
```diff
@@ -1752,6 +1743,67 @@ public SubscribedAPI getSubscriptionById(int subscriptionId) throws APIManagemen
...
+        return null;
+    }
+
+    /** returns the SubscribedAPI object which is related to the UUID
+     *
+     * @param uuid UUID of Application
+     * @return
+     * @throws APIManagementException
+     */
+    public SubscribedAPI getSubscriptionByUUID(String uuid) throws APIManagementException {
```
Comment (sinthuja) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: Too long method, please break into readable logical methods.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java [hunk 3]
```diff
@@ -1825,6 +1825,17 @@ public SubscribedAPI getSubscriptionById(int subscriptionId) throws APIManagemen
         return subscribedAPI;
     }
 
+    /** returns the SubscribedAPI object which is related to the UUID
+     *
+     * @param uuid UUID of Subscription
+     * @return
+     * @throws APIManagementException
+     */
+    public SubscribedAPI getSubscriptionByUUID(String uuid) throws APIManagementException {
```
Comment (Kishanthan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java]: JavaDocs for the return statement on this method is missing. A public API should always have a proper Java Docs.

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIConsumer.java [hunk 4]
```diff
@@ -146,7 +146,15 @@
...
     public SubscribedAPI getSubscriptionById(int subscriptionId) throws APIManagementException;
-    
+
+    /** returns the SubscribedAPI object which is related to the UUID
+     *
+     * @param uuid UUID of Subscription
+     * @return
+     * @throws APIManagementException
+     */
+    public SubscribedAPI getSubscriptionByUUID(String uuid) throws APIManagementException;
```
Comment (Kishanthan) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIConsumer.java]: JavaDocs for the return statement on this method is missing. A public API should always have a proper Java Docs. Fix these issue across all the PR's.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java (hunk 3 above)
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java]: Please note this method was moved and already fixed in 59cfb8da4853453c203780ba4479fae32fbe603c

File: components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/SubscriptionsApiServiceImpl.java (hunk 1 above)
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/SubscriptionsApiServiceImpl.java]: Please note this is already fixed in 908c7a03499e8b8b3567e487a9a7c92a68795144

**Pull request merged**